src/main/java/com/google/graphgeckos/dashboard/datatypes/BuildBotData.java

```diff
@@ -76,6 +80,63 @@ public BuildBotData(String commitHash, String name, List<Log> logs, BuilderStatu
     this.status = status;
   }
 
+  public BuildBotData(@JsonProperty("builderName") String name) {
+    this.name = name;
+  }
+
+  /**
+   * Extracts nested commitHash("revision"), branch("branch") and
+   * timestamp("when") fields from the json
+   *
+   * @param sourceStamp Representation of the json component, where the commitHash field is located
+   */
+  @JsonProperty("sourceStamp")
+  @SuppressWarnings("unchecked")
+  public void unpackSourceStamp(Map<String, Object> sourceStamp) {
+    commitHash = sourceStamp.get("revision").toString();
+    List<Object> changes = (ArrayList<Object>)sourceStamp.get("changes");
+    Map<String, Object> latestChange = (Map<String, Object>)changes.get(0);
+    timestamp = Timestamp.ofTimeMicroseconds(Long.parseLong(latestChange.get("when").toString()));
+  }
+
+  /**
+   * Defines builder status based on a phrase provided in parsed json.
+   * If something failed or the phrase doesn't contain any of the key words
+   * ("failed", "successful", "lost"), then the buildbot is considered failed {@code FAILED}.
+   * If something is lost, then the buildbot is considered lost {@code LOST}.
+   * If  the buildbot is considered passed {@code PASSED}.
+   *
+   * @param words Words of the "text" JSON field
+   */
+  @JsonProperty("text")
+  private void extractStatus(List<String> words) {
+    for (String word : words) {
+      if (word.equals("failed")) {
+        status = BuilderStatus.FAILED;
+        break;
+      }
+      if (word.equals("lost")) {
+        status = BuilderStatus.LOST;
+        break;
+      }
+      if (word.equals("successful")) {
+        status = BuilderStatus.PASSED;
+      }
+    }
+  }
+
+  /**
+   * Unpacks logs represented as list of lists of two strings,
+   * where the first one is a type of the log and the second one
+   * is a link to the log.
+   *
+   * @param logs Representation of the json component, where the logs are located
+   */
+  @JsonProperty("logs")
+  private void unpackLogs(List<String[]> logs) {
```

Comment:
A list of arrays is an usual data structure. Do you get that from some library?

Reply:
No, I thought that the String[] is an Object so it could be used as a type of a generic. But I still replaced it by the List<List<String>>